SU2_CFD/src/solvers/CEulerSolver.cpp

```diff
@@ -7114,6 +7114,69 @@ void CEulerSolver::BC_Inlet(CGeometry *geometry, CSolver **solver_container,
           break;
       }
 
+      /*--- Check if the inlet node is shared with a viscous wall. ---*/
+
+      if (geometry->nodes->GetViscousBoundary(iPoint)) {
+
+        switch (Kind_Inlet) {
+
+          /*--- Total properties have been specified at the inlet. ---*/
+
+          case INLET_TYPE::TOTAL_CONDITIONS: {
+
+            /*--- Impose the wall velocity from the interior wall node. ---*/
+
+            Velocity2 = 0.0;
+            for (iDim = 0; iDim < nDim; iDim++) {
+              V_inlet[iDim+1] = nodes->GetVelocity(iPoint,iDim);
+              Velocity2 += V_inlet[iDim+1] * V_inlet[iDim+1];
+            }
+
+            /*--- Match the pressure, density and energy at the wall. ---*/
+
+            Pressure = nodes->GetPressure(iPoint);
+            Density = Pressure / (Gas_Constant * Temperature);
+            Energy = Pressure / (Density*Gamma_Minus_One) + 0.5 * Velocity2;
+
+            if (tkeNeeded) Energy += GetTke_Inf();
+
+            V_inlet[0] = Temperature;
+            V_inlet[nDim+1] = Pressure;
+            V_inlet[nDim+2] = Density;
+            V_inlet[nDim+3] = Energy + Pressure / Density;
+            break;
+          }
+
+          case INLET_TYPE::MASS_FLOW: {
+
+            /*--- Impose the wall velocity from the interior wall node. ---*/
```

Comment:
I don't understand why the treatment differs for total conditions and mass flow.